pkg/apis/clusteroperator/types.go

```diff
@@ -276,8 +276,7 @@ type ClusterStatus struct {
 	Provisioned bool
 
 	// ProvisionedJobGeneration is the generation of the cluster resource used to
-	// to generate the latest completed infra provisioning job. The value will be set
-	// regardless of the job having succeeded or failed.
+	// generate the latest successful infra provisioning job.
```

Comment:
Do we still have plans to cease starting new jobs after they fail for a long amount of time?

Reply:
Yes but would we not track that explicitly in the owner's status instead of here?

Reply:
This is the owner's status. I don't explicitly have an objection to tracking it elsewhere. Do you have a suggestion about what that would look like?

Reply:
I would repurpose the Failed condition to label the provisioning/install as failed. It really serves no purpose right now since we're starting a new job every time the job fails anyway.

Reply:
Would you expect the controller to clear the condition when it notices a change in the generation? If not, then we would not have a way to differentiate between a failed state from a previous generation as opposed to a failed state from the current generation.

If we did that, then I suggest that we change the semantics of the job generation to be an observed generation that is set as soon as the controller sees the new generation--as opposed to setting the generation when the job succeeds.

In any event, we can address that later when we do work for stopping jobs.

Reply:
Here are some more details about my proposal.

- If the controller sees generation != observedGeneration, the controller clears out job succeeded/failed conditions and records the observedGeneration as the current generation.
- If the controller sees generation == observedGeneration, the controller also looks at job succeeded/failed conditions to determine whether there is work that needs to be done. If neither succeeded/failed conditions are true, then there is work to do.

Reply:
Sounds good to me.